### PR TITLE
Docker-driven tests with latest SDKs

### PR DESCRIPTION
**Pull request by quic-amitraj**
1. Migration to Docker: All tests are now executed within Docker containers, ensuring they utilize the ``latest SDKs``.

2. Automated Container Management: For each PR, Jenkins will create a ``dedicated Docker container``, which will be ``removed`` after the tests are completed.

3. Conflict Avoidance: To prevent conflicts, Docker containers will be named using the ``branch`` or ``PR`` name.

4. Post-Test Cleanup: Directories will be ``cleaned`` up after test execution, regardless of ``pass`` or ``fail`` status.

### CODE REVIEW

File: scripts/Jenkinsfile [hunk 1]
```diff
@@ -1,47 +1,55 @@
...
+                {
+
+                    sh '''
+                    . ~/.bashrc
+                    docker run --privileged -dit --name ${BRANCH_NAME} -v ./:/efficient-transformers -v  ${HF_PATH}:${DOCKER_HF_PATH} ${DOCKER_LATEST}:master_latest
+                    docker exec ${BRANCH_NAME} bash -c "
+                    cd /efficient-transformers &&
+                    . /opt/qeff-env/bin/activate &&
+                    pip install --upgrade pip setuptools &&
+                    pip install .[test] --extra-index-url https://download.pytorch.org/whl/cpu &&
```
Comment (quic-rishinr) [inline scripts/Jenkinsfile:27]: Do we still need to have extra-index-url? we removed it in #139

File: scripts/Jenkinsfile [hunk 2]
```diff
@@ -50,9 +58,24 @@ pipeline
...
          {
-             junit 'tests/tests_log.xml'
-             echo 'Cleaning Workspace'
-             deleteDir() 
+            script
+            {
+                try 
+                {
+                    sh '''
+                    docker rm -f ${BRANCH_NAME}
```
Comment (quic-rishinr) [inline scripts/Jenkinsfile:66]: Since the docker is specific to SDK do we need to docker image as well?

File: scripts/Jenkinsfile (hunk 1 above)
Comment (quic-amitraj) [reply scripts/Jenkinsfile]: Updated to align with 139.

File: scripts/Jenkinsfile (hunk 2 above)
Comment (quic-amitraj) [reply scripts/Jenkinsfile:66]: A new jenkins pipeline is setup to fetch docker with latest SDK nightly.

File: scripts/Jenkinsfile [hunk 3]
```diff
@@ -1,47 +1,55 @@
...
-                '''
+            {                
+                script
+                {
+
+                    sh '''
+                    . ~/.bashrc
+                    docker run --privileged -dit --name ${BRANCH_NAME} -v ./:/efficient-transformers -v  ${HF_PATH}:${DOCKER_HF_PATH} ${DOCKER_LATEST}:master_latest
+                    docker exec ${BRANCH_NAME} bash -c "
+                    cd /efficient-transformers &&
```
Comment (ochougul) [inline scripts/Jenkinsfile:24]: Also set QEFF_HOME value to a local docker path, it's okay if this gets deleted when docker container is deleted.

**Review by quic-rishinr — APPROVED** [review]

**Comment by ochougul** [comment]
Please rebase.

**Comment by quic-amitraj** [comment]
Duplicate #180

**Pull request closed without merging**